src/lib.rs

```diff
@@ -170,7 +170,21 @@ impl AutoCfg {
         // so for now we only apply RUSTFLAGS when cross-compiling an artifact.
         //
         // See https://github.com/cuviper/autocfg/pull/10#issuecomment-527575030.
-        let rustflags = if env::var_os("TARGET") != env::var_os("HOST") {
+        let dir_contains_target = dir
+            .to_str()
+            .and_then(|dir| {
+                env::var("TARGET").ok().map(|target| {
+                    let mut cargo_target_dir =
+                        env::var("CARGO_TARGET_DIR").unwrap_or_else(|_| "target".to_string());
+                    cargo_target_dir.push(MAIN_SEPARATOR);
+                    cargo_target_dir.push_str(&target);
```

Comment:
I think it would be cleaner to build this with `PathBuf::push`.

There are also now 3 places reading TARGET in short order -- we should probably just save that to a local once. Your chain here would then start with `target.as_ref().and_then(...)`.